### PR TITLE
Update Readme_functions

### PR DESCRIPTION
**Pull request by xbreizh**
I realized that some details where missing in that procedure and added them in an attempt to make it clearer.

### CODE REVIEW

File: job-service-db/README_functions.md [hunk 1]
```diff
@@ -8,24 +8,63 @@ differ)
...
+- Ensure that we have `/* Forward Declaration */` in the body of the forward declared function.
+
+If working on a `plpgsql` function, declare the body as:  
+`AS 'BEGIN /* Forward Declaration */ END';`  
+
+Ex:  
+
+    DO $$
+    BEGIN
+    CREATE FUNCTION internal_get_task_status(
```
Comment (dermot-hardy) [inline job-service-db/README_functions.md]: Add some indenting here.  Indent all the rows between `BEGIN` and `EXCEPTION`.

File: job-service-db/README_functions.md [hunk 2]
```diff
@@ -8,24 +8,63 @@ differ)
...
+    in_task_table_name VARCHAR(63)
+    )
+    RETURNS TABLE(
+    status job_status,
```
Comment (dermot-hardy) [inline job-service-db/README_functions.md]: And indent the parameters and the return columns another notch.

File: job-service-db/README_functions.md [hunk 3]
```diff
@@ -8,24 +8,63 @@ differ)
...
     DO $$
     BEGIN
-        CREATE FUNCTION internal_resolve_status(
-        in_current_job_status job_status,
-        in_proposed_job_status job_status
-        )
-        RETURNS job_status
-        LANGUAGE SQL IMMUTABLE
-        AS '/* Forward Declaration */SELECT NULL::job_status';
+    CREATE FUNCTION internal_get_prereq_job_id_options(job_id_with_opts VARCHAR(128))
```
Comment (dermot-hardy) [inline job-service-db/README_functions.md]: Same

File: job-service-db/README_functions.md [hunk 4]
```diff
@@ -8,24 +8,63 @@ differ)
...
 
+## Dropping a function
+When dropping a function, we update the main migration script with the `DROP` function and remove the _function.sql_ file.  
+Ex: `DROP FUNCTION IF EXISTS internal_report_task_completion(in_task_table_name VARCHAR(63));`
+
 ## Updating a function requiring a Change of signature
 
 If we need to change the function signature, then we need to drop the old one. To proceed, we add the drop function element into the 
-main migration script, then update the function in the corresponding script.
+main migration script, then add the new signature as a _forward declared_ function, then update the function in the corresponding
```
Comment (dermot-hardy) [inline job-service-db/README_functions.md]: It sounds like you're saying that all function updates where the signature changes should be forward declared.  I don't think that's right.  It's only if their internal functions - same as newly introduced functions.

**Review by dermot-hardy — APPROVED** [review]

File: job-service-db/README_functions.md (hunk 1 above)
Comment (xbreizh) [reply job-service-db/README_functions.md]: Done

File: job-service-db/README_functions.md (hunk 2 above)
Comment (xbreizh) [reply job-service-db/README_functions.md]: Done

File: job-service-db/README_functions.md (hunk 3 above)
Comment (xbreizh) [reply job-service-db/README_functions.md]: Done

File: job-service-db/README_functions.md (hunk 4 above)
Comment (xbreizh) [reply job-service-db/README_functions.md]: True, reorganized

**Review by dermot-hardy — APPROVED** [review]

**Pull request merged**